sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/parquet/ParquetSchemaPruning.scala

```diff
@@ -280,7 +280,7 @@ private[sql] object ParquetSchemaPruning extends Rule[LogicalPlan] {
           val leftFieldType = leftStruct(fieldName).dataType
           val rightFieldType = rightStruct(fieldName).dataType
           val sortedLeftFieldType = sortLeftFieldsByRight(leftFieldType, rightFieldType)
-          StructField(fieldName, sortedLeftFieldType)
+          StructField(fieldName, sortedLeftFieldType, nullable = leftStruct(fieldName).nullable)
```

Comment:
@viirya, isn't `right` the schema of provided source? Yes, it looks we should use `left` to project it later with correct nullability if I didn't misread but looks logically it's going to set different nullability at `hadoopFsRelation.copy(dataSchema = prunedDataSchema)` later.

I guess it won't be a big deal as long as it works but looks better we clean here as well while we're here.

Reply:
The `right` and `left` should have the same nullability as `left` is pruned version of `right`. As here it sorts the fields in `left` so I use `left` nullability.

Reply:
If left and right have different nullability, it means we do something wrong when preparing pruned schema.

Reply:
Oh, right. LGTM